src/v0/destinations/am/transform.js

```diff
@@ -702,6 +702,12 @@ const processSingleMessage = (message, destination) => {
       logger.debug('could not determine type');
       throw new InstrumentationError('message type not supported');
   }
+  // putting a check for evType as there some cases like page name is not present and teamplte depends on page name only.
+  if (!isDefinedAndNotNull(evType)) {
+    throw new InstrumentationError(
+      'Event type not present. If you are sending page/screen call with custom name, please provide the name',
+    );
+  }
```

Comment:
Please move this to a separate function. This method is already huge.

Reply:
Done